cypress-io/cypress
# feat: Add button to clear value from search fields

- Closes #21686 

### User facing changelog
* Adds new button to clear search field when filtering specs
* Fix styling issue if a very long search value is input on the Specs List page

### Additional details
Refactoring the `Input` component to rely on flexbox for layout of prefix/suffix, was previously using absolute positioning. This caused issues with the filter field in SpecsList- the search value would underlap the suffix section because it had a dynamic width based on the number of specs

### Steps to test
1. Open project into E2E or CT mode
2. On Specs List view, type a value into the search field. Note that a new "X" button appears on the right side immediately left of the result count section
3. Click the "X" button. Observe field is cleared, spec list resets, and "X" button hides
4. Verify that new "X" button is navigable by keyboard and has appropriate `aria-label`
5. Input a very long value into the field, observe that the value scrolls rather than underlaps result count section
6. Navigate into a spec, open inline spec search
7. Type a value into the search field, observe new "X" button appears on right side
8. Click "X" button. Observe field is cleared, spec list resets, and "X" button hides
9. Verify that new "X" button is navigable by keyboard and has appropriate `aria-label`
10. Verify other usages of the `Input` component still style appropriately. Examples:
  * Project filter field in global mode
  * Filename field in "New Spec" modal
 
### How has the user experience changed?

https://user-images.githubusercontent.com/11482842/172703636-237745a7-48bd-4ddb-ae8f-403c749446d3.mov


### PR Tasks
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
